cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -1878,17 +1879,29 @@ TYPED_TEST(TestStringKernels, Strptime) {
   this->CheckUnary("strptime", input4, unit, output4, &options);
 
   options.format = "%m/%d/%Y %%z";
-  this->CheckUnary("strptime", input5, unit, output1, &options);
+  #ifndef EMSCRIPTEN
+    // emscripten bug https://github.com/emscripten-core/emscripten/issues/20466
+    this->CheckUnary("strptime", input5, unit, output1, &options);
 
-  options.error_is_null = false;
-  this->CheckUnary("strptime", input5, unit, output1, &options);
+    options.error_is_null = false;
+    this->CheckUnary("strptime", input5, unit, output1, &options);
 
   EXPECT_RAISES_WITH_MESSAGE_THAT(
       Invalid, testing::HasSubstr("Invalid: Failed to parse string: '5/1/2020'"),
       Strptime(ArrayFromJSON(this->type(), input1), options));
+
+  #else
+    GTEST_SKIP()<< "Skipping some strptime tests due to emscripten bug https://github.com/emscripten-core/emscripten/issues/20466";
+  #endif
```

Comment:
Could you put the following at the beginning instead of `#infdef #else #endif`?

```cpp
  #ifdef EMSCRIPTEN
    GTEST_SKIP()<< "Skipping some strptime tests due to emscripten bug https://github.com/emscripten-core/emscripten/issues/20466";
  #endif
```